webapp-django/crashstats/crashstats/tests/test_views.py

```diff
@@ -1894,6 +1896,339 @@ def mocked_get(url, params, **options):
+            if '/products/build_types/' in url:
+                return Response({
+                    'hits': {
+                        'release': 0.1,
+                        'beta': 1.0,
+                    }
+                })
+
+            raise NotImplementedError(url)
+
+        rget.side_effect = mocked_get
+
+        def mocked_supersearch_get(**params):
+            eq_(params['product'], ['WaterWolf'])
+            eq_(params['version'], ['20.0', '19.0'])
+            end_date = timezone.now().date()
+            start_date = end_date - datetime.timedelta(weeks=2)  # the default
+            expected_dates = [
+                start_date.strftime('>=%Y-%m-%d'),
+                end_date.strftime('<%Y-%m-%d'),
+            ]
+            eq_(params['date'], expected_dates)
+            eq_(params['_histogram.date'], ['version'])
+            eq_(params['_facets'], ['version'])
+            eq_(params['_results_number'], 0)
+            eq_(params['_columns'], ('date', 'version', 'platform', 'product'))
+            assert params['_fields']
+
+            response = {
+                'facets': {
+                    'histogram_date': [],
+                    'signature': [],
+                    'version': []
+                },
+                'hits': [],
+                'total': 21187
+            }
+            totals = {
+                '19.0': 0,
+                '20.0': 0,
+            }
+            while start_date < end_date:
+                counts = dict(
+                    (version, random.randint(0, 100))
+                    for version in ['20.0', '19.0']
+                )
+                date = {
+                    'count': sum(counts.values()),
+                    'facets': {
+                        'version': [
+                            {'count': v, 'term': k}
+                            for k, v in counts.items()
+                        ]
+                    },
+                    'term': start_date.isoformat()
+                }
+                for version, count in counts.items():
+                    totals[version] += count
+                response['facets']['histogram_date'].append(date)
+                start_date += datetime.timedelta(days=1)
+            response['facets']['version'] = [
+                {'count': v, 'term': k}
+                for k, v in totals.items()
+            ]
+            return response
+
+        SuperSearch.implementation().get.side_effect = mocked_supersearch_get
+
+        response = self.client.get(url, {
+            'p': 'WaterWolf',
+            'v': ['20.0', '19.0']
+        })
+        eq_(response.status_code, 200)
+
+        # There's not a whole lot we can easily test in the output
+        # because it's random numbers in the mock functions
+        # and trying to not use random numbers and looking for
+        # exact sums or whatnot is error prone because we might
+        # asset that the number appears anywhere.
```

Comment:
Or you can use ``pyquery`` and select just the right elements to check. :) I find it weird that you use ``random`` in unit tests to be honest, it feels wrong because you cannot test that the numbers are correct.